ambry-router/src/main/java/com.github.ambry.router/AdaptiveOperationTracker.java

```diff
@@ -37,15 +37,17 @@
  * perceived latencies.
  */
 class AdaptiveOperationTracker extends SimpleOperationTracker {
-  static final long MIN_DATA_POINTS_REQUIRED = 1000;
-
+  private final RouterConfig routerConfig;
   private final Time time;
   private final double quantile;
-  private final Histogram localColoTracker;
-  private final Histogram crossColoTracker;
+  private final Histogram localColoHistogram;
+  private final Histogram crossColoHistogram;
   private final Counter pastDueCounter;
   private final OpTrackerIterator otIterator;
   private Iterator<ReplicaId> replicaIterator;
+  private Map<PartitionId, Histogram> localColoPartitionToHistogram;
```

Comment:
Unfortunately, it cannot be `final` here because `localColoPartitionToHistogram` is initialized on demand (That is, it depends on router config and may not be initialized if this is Datacenter level tracker)